src/index.ts

```diff
@@ -8,13 +8,21 @@ export { Redis } from "ioredis";
 
 // CONSTANTS
 const isPublisherInstance = (instance: Instance) => instance === "publisher";
-let publisher: Redis;
-let subscriber: Redis;
+let publisher: Redis | undefined;
+let subscriber: Redis | undefined;
 
 export type Instance = "subscriber" | "publisher";
 
 export function getRedis(instance: Instance = "publisher") {
-  return isPublisherInstance(instance) ? publisher : subscriber;
+  const instanceIsDefined = (instance: Redis | undefined): instance is Redis => typeof instance !== "undefined";
```

Comment:
Peut être pas nécessaire de faire un `is` ici vu que tu ne l'utilise qu'une fois ? Juste `!redis` ne suffirait pas ?